.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: xshop-web-ci
+
+on:
+  push:
+    branches: [develop, master]
+  pull_request:
+    branches: [develop, master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.8]
+    services:
+      postgres:
+        image: postgres:12
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: github_actions
+        ports:
+          - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: psycopg2 prerequisites
+        run: sudo apt-get install python-dev libpq-dev
+      - name: Install Poetry
+        uses: Gr1N/setup-poetry@v3
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pypoetry/virtualenvs
+          key: ${{ runner.os }}-poetry-${{ hashFiles('poetry.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-poetry-
+      - name: Get poetry version
+        run: poetry --version
+      - name: Install dependencies
+        run: poetry install
+      - uses: harmon758/postgresql-action@v1
+        with:
+          postgresql version: "12"
+      - name: Run Migrations
+        run: poetry run python manage.py migrate
+      - name: Run Tests
+        run: poetry run python manage.py test
+        env:
+          DJANGO_SECRET_KEY: ${{ secrets.DJANGO_SECRET_KEY }}
+      - name: Run flake8 linter based on black
+        run: poetry run flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```

Comment:
run flake8 without options, they're defined in the `.flake8` file.
if you want to add more, put them there